autonomy/cli/analyse.py

```diff
@@ -97,8 +97,23 @@ def check_abci_app_specs(
 ) -> None:
     """Check ABCI app specs."""
 
+    # The command expects 'packages_dir' to be named 'packages', so to make import statements to be compatible with
+    # the standard Python import machinery. The directory can be outside the working directory from which the command
+    # is executed.
+    packages_dir = Path(packages_dir).absolute()
+
+    # add parent directory to Python system path
+    # so to give priority to the import of packages modules
+    sys.path.insert(0, str(packages_dir.parent))
```

Comment:
This looks distasteful and I highly recommend against it. This will presumably affect imports in other modules that do not anticipate such wickedness. Lasting state changes should be avoided at all cost. Use a context manager. What's more is that I don't think it should be here; how is this a concern of the `check_abci_app_spec` functionality? It should not be.

Reply:
this is a Click command that is run in a separate Python interpreter, and all the execution paths die with `sys.exit` in any possible execution, so there are no dangerous side effects. But I will add the context manager here so as to be cleaner (I would not add it in `SpecCheck` methods as they should, and do, implicitly assume that the AEA modules in packages are importable using standard Python import machinery).

Reply:
that is very useful to know, more so than the current comment in the code, I suggest exchanging current code comment with the first sentence of your reply.